### PR TITLE
improvement(auth): verbose error message when GitHub repo unavailable

### LINKED ISSUE #545: Add more descriptive repo "Not Found" error.
**- Do you want to request a *feature* or report a *bug*?**
bug?

**- What is the current behavior?**
If `backend.repo` is set to a non-existent repo, a very unclear error message will be shown on login: "Not Found".

**- If the current behavior is a bug, please provide the steps to reproduce.**
- Set the `backend.repo` in your config to a non-existent repo.
- Try to login.

**- What is the expected behavior?**
A clearer error message, maybe "Repo Not Found".

**- Please mention your node.js, and operating system version.**

### PR DESCRIPTION
**Pull request by erquhart**
**Summary**

Provides a descriptive error message in the UI when a repo is not found during login. Fixes #545.

### CODE REVIEW

File: packages/netlify-cms-backend-bitbucket/src/API.js [hunk 1]
```diff
@@ -46,7 +46,14 @@ export default class API {
     ])(req);
 
   user = () => this.request('/user');
-  hasWriteAccess = () => this.request(this.repoURL).then(res => res.ok);
+
+  hasWriteAccess = async () => {
+    const response = await this.request(this.repoURL);
+    if (response.status === 404) {
+      throw Error('Repo not found');
```
Comment (erquhart) [inline packages/netlify-cms-backend-bitbucket/src/API.js]: The bitbucket backend doesn't throw if the repo is not found, so we check for an explicit 404 status on the response.

File: packages/netlify-cms-core/src/actions/auth.js [hunk 2]
```diff
@@ -76,6 +76,7 @@ export function loginUser(credentials) {
         dispatch(authenticate(user));
       })
       .catch(error => {
+        console.error(error);
```
Comment (erquhart) [inline packages/netlify-cms-core/src/actions/auth.js]: We should always log these in the console as well.

File: packages/netlify-cms-core/src/components/UI/Toast.js [hunk 3]
```diff
@@ -9,6 +9,7 @@ injectGlobal`
 
   .notif__container {
     z-index: 10000;
+    white-space: pre-wrap;
```
Comment (erquhart) [inline packages/netlify-cms-core/src/components/UI/Toast.js]: Respect multiline formatting in the notification.

**Comment by verythorough** [comment]
Deploy preview for *netlify-cms-www* ready!

Built with commit 34fb979e5575a1eee0826da4a202061b470cabcd

https://deploy-preview-1831--netlify-cms-www.netlify.com

**Comment by verythorough** [comment]
Deploy preview for *cms-demo* ready!

Built with commit 61b3721eac453fa8ba9ae86c8c9fe75d67cb4354

https://deploy-preview-1831--cms-demo.netlify.com

**Pull request merged**